### PR TITLE
add docs for `ini_parse_quantity` function

### PR DESCRIPTION
**Pull request by realFlowControl**
Hello there :vulcan_salute:

this will add the documentation for the `ini_parse_quantity` function as found in #1803 

Kind regards
Florian

### CODE REVIEW

**Comment by Girgias** [comment]
Just landed [0f8b9eb49bd644f07175e9191d2cc146737446ad](https://github.com/php/php-src/commit/0f8b9eb49bd644f07175e9191d2cc146737446ad) to add support for ``0b`` and ``0o`` prefixes, wondering if it makes sense to add examples with prefixes in the docs? (hex has worked since forever)

File: reference/info/functions/ini-parse-quantity.xml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  &reftitle.parameters;
+  <para>
+   <variablelist>
+    <varlistentry>
+     <term><parameter>shorthand</parameter></term>
+     <listitem>
+      <para>
+       The ini shorthand to parse, must be a number followed by an optional multiplier.
+
+       Multiplier can be any of <literal>k</literal>/<literal>K</literal> (1024),
```
Comment (devnexen) [inline reference/info/functions/ini-parse-quantity.xml:30]: I would suggest
`The following multipliers are supported :
  <literal>k</literal>/<literal>K</literal> (1024),
...
`

File: reference/info/functions/ini-parse-quantity.xml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    </varlistentry>
+   </variablelist>
+  </para>
+ </refsect1>
+
+ <refsect1 role="returnvalues">
+  &reftitle.returnvalues;
+  <para>
+   Returns the interpreted size in bytes as &integer; on success, or
+   <literal>0</literal> in case of unparsable shorthand and raises a
```
Comment (devnexen) [inline reference/info/functions/ini-parse-quantity.xml:44]: `raises an <constant>E_WARNING</constant> level error` or something approaching

File: reference/info/functions/ini-parse-quantity.xml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+   <variablelist>
+    <varlistentry>
+     <term><parameter>shorthand</parameter></term>
+     <listitem>
+      <para>
+       The ini shorthand to parse, must be a number followed by an optional multiplier.
+
+       Multiplier can be any of <literal>k</literal>/<literal>K</literal> (1024),
+       <literal>m</literal>/<literal>M</literal> (1048576),
+       <literal>g</literal>/<literal>G</literal> (1073741824).
```
Comment (cmb69) [inline reference/info/functions/ini-parse-quantity.xml:32]: I suggest to mark-up the numbers:
```suggestion
       Multiplier can be any of <literal>k</literal>/<literal>K</literal> (<literal>1024</literal>),
       <literal>m</literal>/<literal>M</literal> (<literal>1048576</literal>),
       <literal>g</literal>/<literal>G</literal> (<literal>1073741824</literal>).
```

**Comment by cmb69** [comment]
> wondering if it makes sense to add examples with prefixes in the docs?

I think it would make more sense, to add some `ini_get()` examples.

File: reference/info/functions/ini-parse-quantity.xml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+Warning: Invalid quantity "10F": unknown multiplier "F", interpreting as "10" for backwards compatibility
+int(10)
+
+]]>
+    </screen>
+   </example>
+  </para>
+ </refsect1>
+
```
Comment (cmb69) [inline reference/info/functions/ini-parse-quantity.xml:90]:
```suggestion
 <refsect1 role="seealso">
  &reftitle.seealso;
  <simplelist>
   <member><function>ini_get</function></member>
  </simplelist>
 </refsect1>
```

**Comment by devnexen** [comment]
Looks good but cc @cmb69

File: reference/info/functions/ini-parse-quantity.xml (hunk 2 above)
Comment (realFlowControl) [reply reference/info/functions/ini-parse-quantity.xml]: :heavy_check_mark:

**Comment by realFlowControl** [comment]
Hey there,
sorry for taking so long to come back to this PR. I think I incorporated all your hints :smile: 
Hope everything is fine, if there is something else, just ping.
@Girgias, @cmb69: I added the prefixes to the docs, I think it should be there for completeness.

File: reference/info/functions/ini-parse-quantity.xml [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+     </listitem>
+    </varlistentry>
+   </variablelist>
+  </para>
+ </refsect1>
+
+ <refsect1 role="returnvalues">
+  &reftitle.returnvalues;
+  <para>
+   Returns the interpreted size in bytes as &integer; on success, or
```
Comment (Girgias) [inline reference/info/functions/ini-parse-quantity.xml:47]:
```suggestion
   Returns the interpreted size in bytes as an &integer; on success, or
```

File: reference/info/functions/ini-parse-quantity.xml [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+   raises a <constant>E_WARNING</constant> and returns a <literal>0</literal>
+   in case of unparsable shorthand.
+  </para>
+  <para>
+   In case of invalid multiplier the function will return the number and raise
+   a <constant>E_WARNING</constant>.
+  </para>
+ </refsect1>
```
Comment (Girgias) [inline reference/info/functions/ini-parse-quantity.xml:55]:
```suggestion
 <refsect1 role="returnvalues">
  &reftitle.returnvalues;
  <para>
   Returns the interpreted size in bytes as an &integer;.
  </para>
 </refsect1>

 <refsect1 role="errors">
  &reftitle.errors;
  <para>
   If the value cannot be parsed, or an invalid multiplier is used, an <constant>E_WARNING</constant> is raised.
  </para>
 </refsect1>
```

File: reference/info/functions/ini-parse-quantity.xml [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+<![CDATA[
+<?php
+
+var_dump(ini_parse_quantity('1024'));
+var_dump(ini_parse_quantity('1024M'));
+var_dump(ini_parse_quantity('512K'));
+var_dump(ini_parse_quantity('0xFFk'));
+var_dump(ini_parse_quantity('Foobar'));
+var_dump(ini_parse_quantity('10F'));
+
```
Comment (Girgias) [inline reference/info/functions/ini-parse-quantity.xml:72]: Maybe add some examples for the prefixes?

**Review by Girgias — CHANGES_REQUESTED** [review]
Not totally sure for the wording of the error section, but warnings should ideally not be mentioned in the return value section (yes the manual is not super consistent on this)

File: reference/info/functions/ini-parse-quantity.xml (hunk 7 above)
Comment (realFlowControl) [reply reference/info/functions/ini-parse-quantity.xml]: Thanks for the hint, I added two additional examples, one with binary and one octal.

File: reference/info/functions/ini-parse-quantity.xml [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+ </refnamediv>
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <type>int</type><methodname>ini_parse_quantity</methodname>
+   <methodparam><type>string</type><parameter>shorthand</parameter></methodparam>
+  </methodsynopsis>
+  <para>
+   Returns the interpreted size in bytes on success from an ini shorthand.
```
Comment (cmb69) [inline reference/info/functions/ini-parse-quantity.xml:16]: I think it makes sense to link to the explanation of ini shorthands:
```suggestion
   Returns the interpreted size in bytes on success from an <link linkend="faq.using.shorthandbytes">ini shorthand</link>.
```

**Review by cmb69 — COMMENTED** [review]
Besides the missing link, this looks good to me.

File: reference/info/functions/ini-parse-quantity.xml [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+     <listitem>
+      <para>
+       Ini shorthand to parse, must be a number followed by an optional multiplier.
+
+       The followig multipliers are supported: <literal>k</literal>/<literal>K</literal> (<literal>1024</literal>),
+       <literal>m</literal>/<literal>M</literal> (<literal>1048576</literal>),
+       <literal>g</literal>/<literal>G</literal> (<literal>1073741824</literal>).
+
+       The number can be a decimal, hex (prefixed with <literal>0x</literal> or <literal>0X</literal>),
+       octal (prefixed with <literal>0o</literal> or <literal>0O</literal>) or binary (prefixed with
```
[the remark this answers is not in the document]
Comment (realFlowControl) [reply reference/info/functions/ini-parse-quantity.xml]: You are right, just added this in docs and example

**Review by cmb69 — APPROVED** [review]
Thank you!

**Review by Girgias — APPROVED** [review]

**Pull request merged**